GIMP build hangs forever with 3.6.0

https://github.com/msys2/MINGW-packages/actions/runs/14042781889/job/39316680160

From what I see if python calls into cygwin binaries things hang forever, or something like that. It happens consistently.

I can reproduce locally. Downgrading to 3.5.7 fixes things.

What am I missing?
```
$ /cygdrive/c/msys64/clang64/bin/ninja
[1/1] Running bison
FAILED: grammar.c
"C:/msys64/usr/bin/bison.EXE"
/usr/bin/bison: missing operand
Try '/usr/bin/bison --help' for more information.
ninja: build stopped: subcommand failed.
$ uname -a
CYGWIN_NT-10.0-19045 HP-Z230 3.6.0-1.x86_64 2025-03-18 17:01 UTC x86_64 Cygwin
```

Thanks for testing!

Hm, I don't know :(. Can you try with `C:/cygwin64/bin/bison.exe` in the file?

```
$ /cygdrive/c/msys64/clang64/bin/ninja
[0/1] Running bison
ninja: build stopped: interrupted by user
$ uname -a
CYGWIN_NT-10.0-26100 desktop 3.6.0-1.x86_64 2025-03-18 17:01 UTC x86_64 Cygwin
```

It can't be the Windows version difference I think, I can also reproduce with cygwin in a Win8.1 VM.

--- a/winsup/cygwin/fhandler/pipe.cc
+++ b/winsup/cygwin/fhandler/pipe.cc
@@ -1158,7 +1158,7 @@ fhandler_pipe::set_pipe_buf_size ()
   status = NtQueryInformationFile (get_handle (), &io, &fpli, sizeof fpli,
 				   FilePipeLocalInformation);
   if (NT_SUCCESS (status))
-    pipe_buf_size = fpli.InboundQuota;
+    pipe_buf_size = fpli.InboundQuota < PIPE_BUF ? PIPE_BUF : fpli.InboundQuota;
 }
 
 int

